pkg/apis/autoscaling/v2beta2/defaults.go

```diff
@@ -48,4 +88,42 @@ func SetDefaults_HorizontalPodAutoscaler(obj *autoscalingv2beta2.HorizontalPodAu
 			},
 		}
 	}
+
+	// if behavior is specified, we should fill all the 'nil' values with the default ones
+	if obj.Spec.Behavior != nil {
```

Comment:
The API spec says that `Behavior == nil` will also set the defaults.  I see the test cases don't cover that, but I think you want to create a value for behavior and then merge the defaults in, right?